tests/layer_tests/pytorch_tests/test_derive_index.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import numpy as np
+import pytest
+import torch
+from pytorch_layer_test_class import PytorchLayerTest
+
+
+class TestDeriveIndex(PytorchLayerTest):
+    def _prepare_input(self):
+        input_data = np.random.randint(0, 2, (64)) > 0
+        return (input_data,)
+
+    def create_model(self, start, stop, step):
+        class prim_derive_index(torch.nn.Module):
+            def __init__(self, start, stop, step) -> None:
+                super().__init__()
+                self.start = start
+                self.stop = stop
+                self.step = step
+                self.a = torch.tensor([1])
+                self.b = torch.tensor([0])
+
+            def forward(self, x):
+                tensor_list = []
+                for idx in range(self.start, self.stop, self.step):
+                    tensor_list.append(torch.where(x[idx], self.a, self.b))
+                return torch.cat(tensor_list)
```

Comment:
here you program the model to simulate `aten::__derive_index`. Does it exist available primitive in PyTorch API? Or how it can happen in custom models?